src/Mod/Part/Gui/TaskAttacher.cpp

```diff
@@ -396,30 +548,35 @@ void TaskAttacher::onSelectionChanged(const Gui::SelectionChanges& msg)
         if (iActiveRef < static_cast<int>(refs.size())) {
             refs[iActiveRef] = selObj;
             refnames[iActiveRef] = subname;
-        } else {
+        }
+        else {
             refs.push_back(selObj);
             refnames.push_back(subname);
         }
 
-        //bool error = false;
-        try {
-            pcAttach->AttachmentSupport.setValues(refs, refnames);
-            updateListOfModes();
-            eMapMode mmode = getActiveMapMode();//will be mmDeactivated, if selected or if no modes are available
-            if(mmode == mmDeactivated){
-                //error = true;
-                this->completed = false;
-            } else {
-                this->completed = true;
+        pcAttach->AttachmentSupport.setValues(refs, refnames);
+
+        if (i == objNames.size() - 1) {
+            // We check for the moed only for the last ref added. This is to avoid unnecessary warnings
+            // when we handle initial selection.
+            try {
+                updateListOfModes();
+                eMapMode mmode = getActiveMapMode();//will be mmDeactivated, if selected or if no modes are available

```

Comment:
```suggestion
                eMapMode mmode = getActiveMapMode();  //will be mmDeactivated, if selected or if no modes are available
```
(I have no idea why the linter is not triggering on this...)